packages/cactus-plugin-ledger-connector-quorum/src/main/typescript/plugin-ledger-connector-quorum.ts

```diff
@@ -268,6 +289,101 @@ export class PluginLedgerConnectorQuorum
         contractJSON.networks[networkId].address,
       );
       this.contracts[contractName] = contract;
+    } else if (
+      !isWeb3SigningCredentialCactusKeychainRef(req.signingCredential)
+    ) {
+      const networkId = await this.web3.eth.net.getId();
```

Comment:
@TonyRowntree Could we invoke here the newly added method below (the one that also serves the separate endpoint for the no keychain option)?

Reply:
Note to self - Error handling doesnt need to be for each signing creditial, if the signing credential keychain then throw general error saying needs to be signed via keychain.

Reply:
Next step - break up each signing endpoint into seperate endpoint, shovel out the endpoint code into common methods. Also removing the big web3 endpoint and invoke the smaller endpoints in the test cases instead.